apriltags_ros/src/apriltag_detector.cpp

```diff
@@ -269,5 +406,130 @@ bool AprilTagDetector::getTransform(std::string t1, std::string t2, tf::Transfor
   }
 }
 
+tf::Transform getPlaneTransform(pcl::ModelCoefficients coeffs, tf::Vector3 xAxis)
+{
+  ROS_ASSERT(coeffs.values.size() > 3);
+
+  double a = coeffs.values[0];
+  double b = coeffs.values[1];
+  double c = coeffs.values[2];
+  double d = coeffs.values[3];
+
+  // Assume plane coefficients are normalized
+  tf::Vector3 position(-a*d, -b*d, -c*d);
+  tf::Vector3 z(a, b, c);
+
+  // Make sure z points "up"
+  if (z.dot( tf::Vector3(0, 0, -1)) < 0)
+  {
+    z = -1.0 * z;
+  }
+
+  tf::Vector3 y = z.cross(xAxis).normalized();
+
+  tf::Matrix3x3 rotation;
+  rotation[0] = xAxis;   // x
+  rotation[1] = y;       // y
+  rotation[2] = z;       // z
+  rotation = rotation.transpose();
```

Comment:
Not my favorite either.  You could do rotation.transpose().getRotation(orientation) to avoid making the rotation variable both r and r^t